Fixes #11363: Added Store button on wiki page

KarishmaVanwari opened the pull request:
<!-- Add a short description about your changes here-->

Fixes #11363 <!--(<=== Add issue number here)-->

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

<!--We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!-->

<!--If tests do fail, click on the red `X` to learn why by reading the logs.-->

<!--Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software -->

<!--Thanks!-->

KarishmaVanwari commented:
Hi @jywarren, @TildaDares, @cesswairimu!

I've added a `Store` button on wiki page so as to link the store and wiki pages, however, I am unable to fetch the data from `links.yml` file for some reason that I can't figure out. The wiki page gives the following result - 

![image](https://user-images.githubusercontent.com/78212650/186227652-fa080bd5-806e-4fa3-809a-8662ce0b84dd.png)


Please help me figure this out. Thanks!

jywarren commented:
Hi Karishma, is it possible you don't have a record in the DB for a wiki page at that URL locally? Maybe try creating that page so you have one just like in the live site?

KarishmaVanwari commented:
@jywarren - The wiki page with this URL exists, it is only when I add these lines of code that the wiki page can't be seen.

jywarren commented:
Hmm, are you running your local server in development mode? If we are getting an error we should be seeing more output. Otherwise could you try looking in your Rails logs to see what happened? Thank you!

jywarren commented:
I see this in the unit tests... could the path to the yaml file be wrong, possibly? 

```

Errno::ENOENT: No such file or directory @ rb_sysopen - #{Rails.root}/config/links.yml
/home/runner/work/plots2/plots2/config/initializers/load_links.rb:1:in `read'
/home/runner/work/plots2/plots2/config/initializers/load_links.rb:1:in `<top (required)>'
/home/runner/work/plots2/plots2/vendor/bundle/ruby/2.7.0/gems/activesupport-5.2.8.1/lib/active_support/dependencies.rb:285:in `load'
```

cesswairimu commented:
@KarishmaVanwari maybe try using `YAML.load(File.read('config/links.yml'))`

cesswairimu commented:
also wrapping with `"` instead of single quotes should work as rails uses double quotes for interpolation

KarishmaVanwari commented:
Hey @jywarren, @cesswairimu --  I tried out your suggestions, but nothing worked. I've fixed this by adding `require 'yaml'`. 

Kindly provide me the links of store associated with the tags so as to upadte it. I've added sample links as of now.

jywarren commented:
So glad you were able to fix it! Yes, sometimes these libraries are not included by default, it gets me often. Reading now!

jywarren commented on app/views/tag/_miniCard.html.erb:
```diff
@@ -46,6 +46,11 @@
     <% elsif !tag.name.include? ':' %>
       <a class="pull-right btn btn-primary requireLogin" href='/subscribe/tag/<%= tag.name %>'><%= translation('tag.show.follow') %></a>
     <% end %>
+    <% if tag.name == "one" %>
```
Ah, here, could we make it completely parametric, that is, could we say, if it matches ANY tag listed in the LINKS file, could we display it? It might be something like:

```ruby
if LINKS.keys.include?(tag.name)
```

I'm not 100% sure that would work, but something like it? Also, this is in the mini cards, i haven't seen that yet... would you mind adding a screenshot? Thank you! I thought the previous location had been in the upper right corner next to the wiki edit button, but this sounds interesting too!

KarishmaVanwari replied:
> ```ruby
> if LINKS.keys.include?(tag.name)
> ```


@jywarren 
This code did not work, but I figured this out by first checking if such a key exists and if so, checking if the sub-key exists using this:

```ruby
if LINKS.has_key?("tags") && LINKS["tags"].has_key?(tag.name)
```

It works now and looks like this - 

![image](https://user-images.githubusercontent.com/78212650/188283947-089252a3-3c7b-4107-8430-fa673fc0a52b.png)

jywarren commented on config/links.yml:
```diff
@@ -0,0 +1,5 @@
+---
+tags:
+  #sample links
+  one: https://publiclab.org/
```
```suggestion
  microscopes: https://store.publiclab.org/products/microscope
  microscope: https://store.publiclab.org/products/microscope
  infragram: https://store.publiclab.org/products/infragram-diy-filter-pack
  baby-legs: https://store.publiclab.org/products/babylegs
  infragram-pi: https://store.publiclab.org/products/infragram-pi-camera
  infragram: https://store.publiclab.org/products/infragram-diy-filter-pack
  simple-air-sensor: https://store.publiclab.org/products/simple-air-sensor
  pi-camera: https://store.publiclab.org/products/raspberry-pi-camera
  spectrometer: https://store.publiclab.org/products/lego-spectrometer-kit
```

jywarren commented on config/links.yml:
```diff
@@ -0,0 +1,5 @@
+tags:
+  #sample links
+  one: https://publiclab.org/
+  two: https://github.com/publiclab/plots2/pull/11364
```
```suggestion
```
